### ISSUE TITLE
Request PEP 387 exception for remove asyncore, asynchat and stmpd in Python 3.11

### ISSUE THREAD

**Issue by vstinner** [python/steering-council#86]
Context:

* https://bugs.python.org/issue28533
* https://github.com/python/cpython/pull/29521

My PR got 5 approvals by core devs and so I merged it, but [Brett wrote on python-dev](
https://mail.python.org/archives/list/python-dev@python.org/message/WRYNHCXYNBV2CMF6UUB6BLOSVN24HHOI/):
> I don't think it should be ignored and the modules got yanked prematurely. A request can be made to the SC, though, to expedite the removal with only one release raising an exception.

So I formally asks for a SC exception for this removal. If it's rejeted, I understand that the module should be restored.

**Comment by vstinner** [comment]
I advertized the removal on What's New in Python 3.11, python-dev and Twitter to make sure that most people are aware of the removal.

* https://docs.python.org/dev/whatsnew/3.11.html#removed
* https://mail.python.org/archives/list/python-dev@python.org/thread/LZOOLX5EKOITW55TW7JQYKLXJUPCAJB4/
* https://twitter.com/VictorStinner/status/1460391333265346564
* https://twitter.com/VictorStinner/status/1458775053139689474

I'm not using Twitter as a poll to take decisions, it's only to attempt to announce the change to as much persons as possible before Python 3.11 is released. So if something goes wrong, there is enough time until "3.11.0 final: Monday, 2022-10-03" to revert the change.

**Comment by vstinner** [comment]
asyncore/asynchat deprecation has a long history starting in 2015: https://bugs.python.org/issue25002

**Comment by hroncok** [comment]
I must say I hope announcing changes on Twitter is not relevant to requests like this.

**Comment by vstinner** [comment]
I searched for ``import (asyncore|asynchat|smtpd)`` in [the source code of PyPI top 5000 project](https://mail.python.org/archives/list/python-dev@python.org/thread/WQVEHLRIVISPFMWSSX5N4TQPIUN2XS22/). At December 1st, 21 projects match this regex and I warned 2 projects. See https://bugs.python.org/issue28533#msg407559 for more details.

**Comment by brettcannon** [comment]
> I must say I hope announcing changes on Twitter is not relevant to requests like this.

I'm not quite sure what you're referring to (which I guess answers your question of how relevant Twitter is 😉)?

**Comment by pablogsal** [comment]
After some exhaustive discussion, the Steering Council has decided to **reject** the proposal to deprecate `asynccore`, `asyncchat` and `stmpd` in Python 3.11. We strongly believe that we should strictly follow the [backward compatibility policy](https://www.python.org/dev/peps/pep-0387). As the deprecation warnings for these modules **have not been active for at least two releases** and we consider that this doesn't qualify for an exception, the Steering Council has decided that the **changes in PR29521 should be reverted** at the earliest convenience.

We sympathize with the fact that these modules have been marked as deprecated for a long time and the arguments exposed, but we want to remark on the importance of following procedure and the importance of good communication and the trust relationship between the core team and users. This has to lead us to decide as described previously.

Please, feel free to reopen the issue if you have questions or you wish to discuss some aspect of the decision.

**Comment by vstinner** [comment]
FYI these modules are removed again in Python 3.12: https://github.com/python/cpython/issues/72719

The removal is now covered by PEP 594, approved by the SC.

> I must say I hope announcing changes on Twitter is not relevant to requests like this.

No, it's not. I communicated on python-dev. I also made an announcement on Twitter to attempt to notify people who are not subscribed to this mailing list. I'm trying to communicate as much as possible on incompatible changes so people have more time before Python 3.12 final release to adapt their code. But these announcements are not a justification for the removal.
